Makefile.include

```diff
@@ -64,10 +64,17 @@ else
 endif
 
 ifeq ($(QUIET),1)
-	AD=@
-	MAKEFLAGS += --no-print-directory
+  AD=@
+  MAKEFLAGS += --no-print-directory
 else
-	AD=
+  AD=
+endif
+
+# Fail on warnings. Can be overridden by `make WERROR=0`.
+WERROR ?= 1
+export WERROR
```

Comment:
`make buildtest`'s call to sub-make didn't see the correct value for the `WERROR` variable without it (I tried it both as a bash variable `$${WERROR}` and as a make variable `$(WERROR)`)